lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1163,6 +1176,8 @@ def deploy_vm(self):
 
         # find matching folders
         if self.params['folder'].startswith('/'):
+            if not self.params['folder'].startswith('/vm'):
```

Comment:
There is a similar thing in getvm function. /vm/ is a masked folder so we add it in path if it's not set.
It's allows to define full path without vmware knowledges.